index.js

```diff
@@ -69,64 +69,78 @@ async function run() {
     configs.repoOptions
   );
 
-  return octokit.paginate(workflowRunsRequest).then(workflowRuns => {
-    const artifactPromises = workflowRuns
-      .filter(workflowRun => {
-        const skipWorkflow =
-          configs.skipTags && taggedCommits.includes(workflowRun.head_sha);
-
-        if (skipWorkflow) {
-          console.log(`Skipping tagged run ${workflowRun.head_sha}`);
-
-          return false;
-        }
-
-        return true;
-      })
-      .map(workflowRun => {
-        const workflowRunArtifactsRequest = octokit.actions.listWorkflowRunArtifacts.endpoint.merge(
-          {
-            ...configs.repoOptions,
-            run_id: workflowRun.id,
-          }
-        );
-
-        return octokit.paginate(workflowRunArtifactsRequest).then(artifacts =>
-          artifacts
-            .filter(artifact => {
-              const createdAt = moment(artifact.created_at);
-
-              return createdAt.isBefore(configs.maxAge);
-            })
-            .map(artifact => {
-              if (devEnv) {
-                return new Promise(resolve => {
-                  console.log(
-                    `Recognized development environment, preventing ${artifact.id} from being removed.`
-                  );
-
-                  resolve();
-                });
-              }
-
-              return octokit.actions
-                .deleteArtifact({
-                  ...configs.repoOptions,
-                  artifact_id: artifact.id,
-                })
-                .then(() => {
-                  console.log(
-                    `Successfully removed artifact with id ${artifact.id}.`
-                  );
-                });
-            })
-        );
+  return octokit
+    .paginate(workflowRunsRequest, ({ data }, done) => {
+      const stopPagination = data.find(workflowRun => {
+        const createdAt = moment(workflowRun.created_at);
+
+        return createdAt.isBefore(moment.utc().subtract(90, "days"));
```

Comment:
Two proposals:
- Shall I add a comment on this expression? Perhaps it's *not* obvious what it's supposed to do at first glance. Another option could be to mention this in the doc and leave a reference on this 90-days expiration.
- Wouldn't it make sense to log the date from which the artifacts are *not* taken into account?

Reply:
I think none of them is needed. Older artifacts are removed automatically anyways.